frontend/src/components/guards/CommonGuard.tsx

```diff
@@ -16,23 +18,27 @@ interface Props {
 /**
  * Applies guards enabled in this order:
  * 1. Checks authenticated
- * 2. Checks onboarded
- * 3. Checks not onboarded
+ * 2. Checks not authenticated
+ * 3. Checks onboarded
+ * 4. Checks not onboarded
  */
 export default function CommonGuard({
   children,
   authenticated = false,
+  notAuthenticated = false,
   onboarded = false,
   notOnboarded = false,
 }: Props) {
   const checkAuth = useCheckAuth();
+  const checkNotAuth = useCheckNotAuth();
   const checkOnboarded = useCheckOnboarded();
   const checkNotOnboarded = useCheckNotOnboarded();
 
   const guards = [];
   authenticated && guards.push(checkAuth);
+  notAuthenticated && guards.push(checkNotAuth);
   onboarded && guards.push(checkOnboarded);
-  notOnboarded && guards.push(checkNotOnboarded);
+  notOnboarded && guards.push(() => checkNotOnboarded(true, true));
```

Comment:
Nit: Prefer named args for check not onboarded

Reply:
Done!